### PR TITLE
Fix for versions list, protective programming added.

### PR DESCRIPTION
**Pull request by dandwal**
### What

Was an issue on environments trying to access a nil value causing the versions-list page to crash. It should look something like this (ignore the dates)
Screenshot 2019-06-21 at 10 24 09

### How to review

Checkout branch creates a version of an edition, then create another version of an edition but add a correction alert to the page. Then navigate to the versions list, reason(s) for change should appear.
 
### Who can review

Anyone except me

### CODE REVIEW

File: mapper/mapper.go [hunk 1]
```diff
@@ -276,9 +276,13 @@ func CreateVersionsList(ctx context.Context, d dataset.Model, edition dataset.Ed
...
+		if *ver.Alerts != nil {
+			for _, alert := range *ver.Alerts {
+				if &alert != nil {
+					if alert.Type == correctionAlertType {
+						correctionReasons = append(correctionReasons, alert.Description)
+					}
+				}
 			}
 		}
 		version.Reasons = correctionReasons
```
Comment (eldeal) [inline mapper/mapper.go]: would you want to assign an empty slice to Reasons if there were no alerts? or could we move it inside the `if *ver.Alerts != nil` block?

File: mapper/mapper.go [hunk 2]
```diff
@@ -276,9 +276,13 @@ func CreateVersionsList(ctx context.Context, d dataset.Model, edition dataset.Ed
 
 		var correctionReasons []string
 		const correctionAlertType = "correction"
-		for _, alert := range *ver.Alerts {
-			if alert.Type == correctionAlertType {
-				correctionReasons = append(correctionReasons, alert.Description)
+		if *ver.Alerts != nil {
+			for _, alert := range *ver.Alerts {
+				if &alert != nil {
+					if alert.Type == correctionAlertType {
```
Comment (eldeal) [inline mapper/mapper.go]: It's more common in Go to collapse these 2 conditions into a single if statement to reduce the indentation as they have no elses to require the separation

File: mapper/mapper.go (hunk 1 above)
Comment (dandwal) [reply mapper/mapper.go]: Moved :)

File: mapper/mapper.go (hunk 2 above)
Comment (dandwal) [reply mapper/mapper.go]: Adjusted :)

**Pull request merged**